print/fonttools/Portfile

```diff
@@ -14,15 +15,37 @@ platforms           darwin
 categories          print
 license             BSD
 maintainers         nomaintainer
-homepage            http://fonttools.sourceforge.net/
-master_sites        sourceforge
 
-checksums           rmd160  392a8b4df387bfd45a572d97839799b1b876176a \
-                    sha256  95e5d8a67e2dae3ea237f9bebc449e0af42f12f03683db34f617e5b15d9d4fb0
+checksums           rmd160  eca06bb1039415631100b54733d4c9bdb7d64071 \
+                    sha256  18baa60c8f59a389883bad681ce022cfca2a35b3f128e868ba500390c15893cb
 
 python.default_version 27
 
-depends_build-append port:py27-numpy
+variant woff2 description {Install dependencies for WOFF 2.0 font support} {
+    depends_lib-append port:py27-brotli
+}
+
+variant sym description {Install dependencies for symbolic font statistics analysis} {
+    depends_lib-append port:py27-sympy
+}
+
+variant cocoa description {Install dependencies for Cocoa glyph drawing} {
+    depends_lib-append port:py27-pyobjc
+}
+
+variant qt description {Install dependencies for Qt glyph drawing} {
+    depends_lib-append port:py27-pyqt5
+}
+
+variant png description {Install dependencies for PNG glyph drawing} {
+    depends_lib-append port:py27-reportlab
+}
+
+variant gui description {Install dependencies for GUI font inspector} {
+    depends_lib-append port:py27-pygtk
+}
+
+variant all requires woff2 sym cocoa qt png gui {Install all available optional dependencies} {}
```

Comment:
Are these just runtime dependencies or are they required at the build time already?

If these are build time dependencies as well, I suspect we'll see opportunistic linking.

How heavy are these dependencies in terms of the number of other dependencies pulled in and in terms of licence conflicts (= inability to get a binary)?

It might be nice to simplify the port (that is: less variants). But if variants are justified, that's ok.

I'm tempted to commit the first part right away and leave some time for discussion about the second part.

Reply:
They are just runtime dependencies.

Licenses and dependencies:

- py27-brotli: MIT / none
- py27-sympy: BSD / small
- py27-pyobjc: MIT / small
- py27-pyqt5: GPL-3 / very big
- py27-reportlab: BSD / medium
- py27-pygtk: LGPL-2.1+ / very big

I don't know off the top of my head the implications of the above mix of licenses, but the Qt and GTK dependencies are quite heavy.

My main intent in adding the variants was user convenience; without the variants it's hard to discover what optional features are available, and which can be enabled within the confines of MacPorts. If you think they should be left out then that's OK with me.

Reply:
Oh, no, I didn't want to suggest removing the variants / dependencies, just trying to figure out if simplifying is an option.

If these are just runtime dependencies, the lines should be changed to reflect that (`depends_run`). The only sad thing is that a variant means that the port needs to be recompiled each time just for the sake of adding an additional dependency which is somewhat suboptimal.

If some dependencies are often useful and small, I would make them default (you can still put them into a variant and thus enable a slimmer installation, but make the variant default). Qt and GTK are a different story of course.

Reply:
> If these are just runtime dependencies, the lines should be changed to reflect that (depends_run).

Thanks, I've done that now.

> If some dependencies are often useful and small, I would make them default 

This is a pretty extensive package, and I'm not an expert, but I think `+woff2` by default would be good, as it's small and webfont subsetting is probably a major use case. The others I'm not so sure about. I've now set `+woff2` by default.